dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokeTelnetHandler.java

```diff
@@ -182,27 +80,17 @@ public String telnet(Channel channel, String message) {
         } catch (Throwable t) {
             return "Invalid json argument, cause: " + t.getMessage();
         }
-        if (paramTypes != null) {
-            if (paramTypes.length != list.size()) {
-                return "Parameter's number does not match the number of parameter class";
-            }
-            List<Object> listOfActualClass = new ArrayList<>(list.size());
-            for (int ii = 0; ii < list.size(); ii++) {
-                if (list.get(ii) instanceof JSONObject) {
-                    JSONObject jsonObject = (JSONObject) list.get(ii);
-                    listOfActualClass.add(jsonObject.toJavaObject(paramTypes[ii]));
-                } else {
-                    listOfActualClass.add(list.get(ii));
-                }
-            }
-            list = listOfActualClass;
-        }
 
         Method invokeMethod = null;
         ProviderModel selectedProvider = null;
         for (ProviderModel provider : ApplicationModel.allProviderModels()) {
```

Comment:
I think can be rewritten as 

```
selectedProvider=ApplicationModel.allProviderModels()
                .stream()
                .filter(provider->isServiceMatch(service, provider))
                .findFirst().orElse(null);
```

In this case we may need to make **service** as final.  What do you say?

Reply:
Looking at isServiceMatch, it looks it has a or condition where it will return **true** if **service** is null or blank. So is it a expected? @chickenlj @beiwei30 could you have a look at this. Because this is an existing behavior?